src/Framework/Framework/Compilation/ControlTree/ControlTreeResolverBase.cs

```diff
@@ -770,9 +758,29 @@ private IAbstractPropertySetter ProcessElementProperty(IAbstractControl control,
                     return treeBuilder.BuildPropertyControl(property, null, propertyWrapperElement);
                 }
             }
+            else if (IsCollectionProperty(property))
+            {
+                var collectionType = GetCollectionType(property);
```

Comment:
```suggestion
                var collectionType = GetElementType(property);
```

and use ReflectionUtils if possible

Reply:
ReflectionUtils can't be used, because this works on the ITypeDescriptor. I don't want to rename the method, since it's method which already existed and is virtual